driver/src/main/java/org/neo4j/driver/internal/cluster/loadbalancing/LoadBalancer.java

```diff
@@ -193,6 +286,21 @@ private BoltServerAddress selectAddress( AccessMode mode, AddressSet servers )
         }
     }
 
+    private BoltServerAddress selectAddressAsync( AccessMode mode, AddressSet servers )
```

Comment:
Do we need this sync/async distinction at this level? Do you have plans to implement selection differently in the future?